modules/4337/hardhat.config.ts

```diff
@@ -33,7 +34,8 @@ import './src/tasks/local_verify'
 import './src/tasks/deploy_contracts'
 import './src/tasks/show_codesize'
 
-const solidityVersion = SOLIDITY_VERSION || '0.8.23'
+const defaultSolidityVersion = '0.8.23'
+const primarySolidityVersion = SOLIDITY_VERSION || defaultSolidityVersion
 const soliditySettings = SOLIDITY_SETTINGS
   ? JSON.parse(SOLIDITY_SETTINGS)
   : {
```

Comment:
I interpreted the issue as "lets turn on `viaIR` optimization flag in the repo (and not just makes sure it works). It _should_ have a positive impact on the gas characteristics of the 4337 module.